### DISCUSSION TITLE
Unable to use protobuf-c generated code

### DISCUSSION THREAD

**Discussion by BharathZoho** [protobuf-c/protobuf-c#639] [Q&A] [answered]
Hi,
I tried writing a redis module which used protobuf. I used the following protobuf (for C)
https://github.com/protobuf-c/protobuf-c. I am able to compile and link the module but when i load the module while starting the redis-server, i get the following message,

Module /Users/bharath-2336/redismodules/module.so failed to load: dlopen(/Users/bharath-2336/redismodules/module.so, 6): Symbol not found: _protobuf_c_message_free_unpacked

**Comment by edmonds** [comment, marked as answer]
`.pb-c.c` files aren't standalone. When compiled they must be linked against the `protobuf-c` library.
